### ISSUE TITLE
Cargo fails to uninstall package with "." in name

### ISSUE THREAD

**Issue by LEARAX** [rust-lang/cargo#5094]
I'm attempting to uninstall a package I got from [here](https://github.com/crespyl/rust_ca). The Cargo.toml file lists the name as "ca.rs", and the binary name as "ca".

I installed it with a simple `cargo install` from its top level directory. I now wish to uninstall it. However, I can't.

`cargo install --list` returns:
```
( ⌁ ) G ) EE ) cargo install --list
ca.rs v0.1.0 (file:///Users/Will/Google%20Drive/EE/rust_ca):
    ca
cargo-edit v0.3.0-beta.1:
    cargo-add
    cargo-rm
    cargo-upgrade
heliometer v0.2.1 (file:///Users/Will/Termix/heliometer):
    helio
racer v2.0.12:
    racer
tealdeer v0.4.0:
    tldr
```
However, `cargo uninstall "ca.rs"` (with or without quotes) returns:
```
( ⌁ ) G ) EE ) cargo uninstall ca.rs                                                                                                                                                                       14:39:26 ≡ 101
error: invalid package id specification: `ca.rs`

Caused by:
  invalid character in pkgid `ca.rs`: `.`
```
I am able to uninstall other packages just fine.

**Comment by JuanPotato** [comment]
I think the name is `ca`
```[[bin]]
name = "ca"
path = "src/ca.rs"
```

**Comment by LEARAX** [comment]
> I think the name is `ca`
```
Will ) cargo install --list
bat v0.7.0:
    bat
ca.rs v0.1.0 (file:///Users/Will/Google%20Drive/Extended_Essay/rust_ca):
    ca
exa v0.8.0:
    exa
fd-find v7.1.0:
    fd
racer v2.1.5:
    racer
ripgrep v0.10.0:
    rg
rusty-tags v2.10.0:
    rusty-tags
tealdeer v1.0.0:
    tldr
Will ) cargo uninstall ca
error: package id specification `ca` matched no packages
Will ) cargo uninstall ca.rs
error: invalid package id specification: `ca.rs`

Caused by:
  invalid character in pkgid `ca.rs`: `.`
```
It still refuses to uninstall.

**Comment by ehuss** [comment]
As of #6021 Cargo no longer allows package names with `.` in it.  I'm going to close this as it's not possible to install them anymore, they were never allowed on crates.io (AFAIK), and should be exceedingly rare otherwise.  If you want to clean up your existing installation, just manually remove the entry from `~/.cargo/.crates.toml` and delete the file.
